src/main/java/androidx/core/widget/ImageView.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.core.widget
+
+import android.graphics.ColorMatrix
+import android.graphics.ColorMatrixColorFilter
+import android.graphics.Paint
+import android.widget.ImageView
+
+/**
+ * Filtering an image based on saturation
+ *
+ * @param saturation Value to set the saturation
+ */
+fun ImageView.filter(saturation: Float) {
```

Comment:
This is way too specific of an API. At least if the method was called `saturation()` it would be more sensible. But even then, it doesn't feel general purpose enough. It would be better to make color matrix/filters easier to create.

Reply:
Thanks @romainguy for your input. it's true, I created this feature based on My problem when filtering an image to grey scale only that take a parameter from saturation. Good insight from you to make this more general, cool!